File: codewars/6kyu/convert-string-to-camel-case/dduck-convert-string-to-camel-case.js
// Convert string to camel case
// https://www.codewars.com/kata/517abf86da9663f1d2000003/

const toCamelCase = (str) => str.split('')
  .reduce((a, c, i) => {
    if (a && a[i - 1].match(/[_,-]/)) {
      return a.slice(0, i) + c.toUpperCase();
    }
    return a + c;
  }, '')
  .replace(/[_,-]/gi, '');


console.log(toCamelCase(''), '', "An empty string was provided but not returned");
console.log(toCamelCase("the_stealth_warrior"), "theStealthWarrior", "toCamelCase('the_stealth_warrior') did not return correct value");
console.log(toCamelCase("The-Stealth-Warrior"), "TheStealthWarrior", "toCamelCase('The-Stealth-Warrior') did not return correct value");
console.log(toCamelCase("A-B-C"), "ABC", "toCamelCase('A-B-C') did not return correct value");

File: codewars/6kyu/convert-string-to-camel-case/dduck-convert-string-to-camel-case.py
# Convert string to camel case
# https://www.codewars.com/kata/517abf86da9663f1d2000003/

from unittest import TestCase


def to_camel_case(text):
    result = []
    for i, v in enumerate(text):
        if not text[i - 1].isalpha():
            result.append(v.upper())
        else:
            result.append(v)

    return "".join(filter(lambda v: v.isalpha(), result))


TestCase().assertEqual(to_camel_case(''), '', "An empty string was provided but not returned")

TestCase().assertEqual(to_camel_case("the_stealth_warrior"), "theStealthWarrior",
                       "to_camel_case('the_stealth_warrior') did not return correct value")
TestCase().assertEqual(to_camel_case("The-Stealth-Warrior"), "TheStealthWarrior",
                       "to_camel_case('The-Stealth-Warrior') did not return correct value")
TestCase().assertEqual(to_camel_case("A-B-C"), "ABC", "to_camel_case('A-B-C') did not return correct value")
